### ISSUE TITLE
False positive unexpected-keyword-arg (E1123)

### ISSUE THREAD

**Issue by Gollam** [pylint-dev/pylint#9922]
### Bug description

```python
# pylint: disable=missing-module-docstring
from pyroute2 import IPRoute

a = IPRoute()
b = a.get_addr(index=0)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:5:4: E1123: Unexpected keyword argument 'index' in method call (unexpected-keyword-arg)

-----------------------------------
Your code has been rated at 0.00/10
```

### Expected behavior

I checked pyroute2 source code, and the get_addr function has a **kwarg parameter: https://github.com/svinota/pyroute2/blob/01f92c84977c20f591cfa02452ff588ad0cc73d5/pyroute2/iproute/linux.py#L538

So it accepts 'index' (and the function works correctly, it filters by index), so that's why I think this is a false positive.

### Pylint version

```shell
pylint 3.2.7
astroid 3.2.4
Python 3.10.12 (main, Jul 29 2024, 16:56:48) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04.4 LTS

### Additional dependencies

```python
pyroute2==0.7.12
```

**Comment by akamat10** [comment]
I wonder if it has to do with the fact that there are four flavors of `IPRoute` and one of the flavors ([IPMock one](https://github.com/svinota/pyroute2/blob/01f92c84977c20f591cfa02452ff588ad0cc73d5/pyroute2/iproute/ipmock.py#L682)) doesn't accept `kwarg`. The different flavors are picked based on the `sys.platform` output [here](https://github.com/svinota/pyroute2/blob/01f92c84977c20f591cfa02452ff588ad0cc73d5/pyroute2/iproute/__init__.py#L145).

**Comment by akamat10** [comment]
Right. This wouldn't be a bug in that case.
